agent/agent_test.go

```diff
@@ -155,6 +157,22 @@ func (s *AgentSuite) SetupTest() {
 	s.ctx = ctx
 	s.tc.logger, err = s.mockCommunicator.GetLoggerProducer(ctx, &s.task, nil)
 	s.NoError(err)
+	s.tc.addTaskCommandCleanups([]internal.CommandCleanup{{
+		Command: "cleanup_command",
+		Run: func(ctx context.Context) error {
+			s.ranCommandCleanupsTask = true
```

Comment:
nit: since this is only used in this one test and no other test needs it, might be easier to just count the number of runs rather than set a bool. For example:
```
Run: func(...) {
s.taskCleanupCount++
}
assert.Equal(t, 1, i, "task cleanup should run exactly once")
```
It's a slightly nicer check since it verifies `Run` runs the exact number of times (and not more).